test/com/mjcode/model/PieceTest.java

```diff
@@ -56,34 +56,116 @@ void testCanCapture() {
 
     @Test
     void testCanMoveToOccupiedTile() {
-        Tile targetTile = new Tile(5,5);
+        Tile targetTile = new Tile(3,4);
         Piece whitePiece = new Piece(player1, whiteCurrentTile, 1);
 
-        assertFalse(targetTile.isOccupied());
+        // Add piece to the target tile
+        targetTile.setOccupant(whitePiece);
+
+        // Test when the target tile is occupied by a checkers piece
+        assertTrue(targetTile.isOccupied());
         assertFalse(whitePiece.canMoveTo(targetTile, board));
     }
 
     @Test
-    @Disabled
     void testCanMoveToLegalMove() {
 
-        Tile targetTile = new Tile(5,5);
+        Tile targetTile = new Tile(3,6);
         Piece whitePiece = new Piece(player1, whiteCurrentTile, 1);
-        targetTile.isOccupied();
 
         assertTrue(whitePiece.canMoveTo(targetTile, board));
-
     }
     @Test
     void testCanMoveToIllegalMove() {
 
+        Tile backwardTile = new Tile(1,6);
+        Piece whitePiece = new Piece(player1, whiteCurrentTile, 1);
+
+        assertFalse(whitePiece.canMoveTo(backwardTile, board));
+
+    }
+    @Test
+    void testPawnCanMoveToCanJump() {
+
+        Tile targetTile = new Tile(4,7);
+        Tile blackTile = new Tile(4, 5);
+
+        Piece whitePiece = new Piece(player1, whiteCurrentTile,1);
+        Piece blackPiece = new Piece(player2, blackTile, -1);
+
+        int rowDiff = targetTile.getRow() - whiteCurrentTile.getRow();
+        int colDiff = targetTile.getColumn() - whiteCurrentTile.getColumn();
+
+        // Get the jumped row and column
+        int jumpedRow = whiteCurrentTile.getRow() + (rowDiff / 2);
+        int jumpedCol = whiteCurrentTile.getColumn() + (colDiff / 2);
+
+        Tile jumpedTile = new Tile(3,6);
+
```

Comment:
Remove blackTile completely. Declare blackPiece here and set its position to jumpedTile.